app/src/main/java/com/victor/adasfeed/FeedActivity.kt

```diff
@@ -27,6 +30,27 @@ class FeedActivity : AppCompatActivity() {
     private lateinit var postAdapter: PostAdapter
     private lateinit var storiesAdapter: StoriesAdapter
     private lateinit var headerView: View
+    private lateinit var tvUsername : TextView
+    private lateinit var tvNickname : TextView
+
+    private val registerActivityResult  =
+        prepareResultProfileActivity()
+
+    private fun prepareResultProfileActivity() =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
```

Comment:
Kudos for bringing this up!
Although we use `registerActivityResult` when we intend to open an acitivity expecting some return, for example: I open another app in order to get some data (the main purpose is to get the data, you know?) and when we return to our activity, we get this data (For example, opening the camera and taking a picture. We return to our acitivity with the image).

In our case, we can get some result, but the main purpose of `ProfileActivity` is to manage the profile data, not pass the datas.

Reply:
That is true. I used it thinking about getting the data from the activity without having to use startActivity since we are not using a database, but I realized that it is a kludge. 
Thanks for your help and i will fix it!! 🚀